api/pkg/transformer/symbol/registry.go

```diff
@@ -195,7 +195,57 @@ func (sr Registry) PolarAngle(latitude1 interface{}, longitude1 interface{}, lat
 
 // ParseTimeStamp convert timestamp value into time
 func (sr Registry) ParseTimestamp(timestamp interface{}) interface{} {
-	ts, err := sr.evalArg(timestamp)
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return time.Unix(ts, 0).In(tz)
+	}
+	// empty timezone mean using UTC
+	return sr.processTimestampFunction(timestamp, "", timeFn)
+}
+
+// IsWeekend check wheter given timestamps falls in weekend, given timestamp and timezone
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) IsWeekend(timestamp interface{}, timezone string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.IsWeekend(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// FormatTimestamp will convert timestamp into formatted date string
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) FormatTimestamp(timestamp interface{}, timezone, format string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
```

Comment:
Wouldn't it be slow if we recreate a new function every time it's being called?

Reply:
No, it's not slower. I put the benchmark result in time_bench_test.go file.